### PR TITLE
Optly segment load

### PR DESCRIPTION
**Pull request by aprildagonese**
### What does this pull request do?
*Briefly summarize any features implemented, bugs fixed, or other code changes.*

- Asynchronously loads:
1. Optimizely SDK
2. Optimizely Web
3. Optimizely Client Instance
4. Segment

#### Required before merge:
- [ x ] Has tests (where applicable)
- [ x ] Has appropriate documentation (comments, readme, etc.)
- [ x ] Passes CircleCI & Heroku builds

### What is the relevant story?
*Link to the story if there is one. If not, briefly explain why this change was needed.*

[[Title of the story]]( https://app.clubhouse.io/guild/story/ )

### How do you feel?
*Attach a GIF that describes how you feel about what you wrote or implemented. Feel free to include additional context!*

### CODE REVIEW

File: public/index.ts [hunk 1]
```diff
@@ -1 +1,103 @@
...
+    loadSegment('initial')
+};
+
+function loadSegment(version:string) {
+    const segmentKey = process.env.SEGMENT_KEY
+
+    function segmentSnippet() {!function(){var analytics=(window as any).analytics=(window as any).analytics||[];if(!analytics.initialize)if(analytics.invoked)(window as any).console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=("https:"===document.location.protocol?"https://":"http://")+`scdn.guildeducation.com/analytics.js/v1/${segmentKey}/analytics.min.js`;var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="3.1.0";
```
Comment (elanalynn) [inline public/index.ts]: Any reason to not have this at the top level scope?

File: public/index.ts [hunk 2]
```diff
@@ -1 +1,103 @@
...
+
+function notABot() {
+    const user:string = window.navigator.userAgent.toLowerCase()
+    const botList:Array<string> = ["uptime.com", "baiduspider", "bingbotremove", "duckduckbot", "exabot", "googlebot", "slurp", "sogou", "yandexbox", "facebookexternalhit", "facebot", "ia_archiver"]
+    return !botList.some(botString=>user.includes(botString))
+}
+
+function loadOptlySDK() {
+    new Promise ((resolve, reject) => {
+        var script = document.createElement('script');
```
Comment (elanalynn) [inline public/index.ts]: Might want to defer to a `const` here since it won't change.

File: public/index.ts [hunk 3]
```diff
@@ -1 +1,103 @@
...
+        script.type = 'text/javascript'
+        script.src = `https://students.guildacceptance.com/optimizelysdk.js`;
```
Comment (elanalynn) [inline public/index.ts]: We'll want to use an environment variable here so we can support prod too!

File: public/index.ts [hunk 4]
```diff
@@ -1 +1,103 @@
...
+function loadDatafile() {
+    const optimizely_url:string = `${process.env.OPTIMIZELY_BASE_URL}/optimizelydatafile=${process.env.OPTIMIZELY_SDK_KEY}.json`;
+    fetch(optimizely_url, { mode: 'cors' })
+    .then((response) => {
+        return response.json()
+    })
+    .then((datafile) => { createClientInstance(datafile) })
+};
+
+var customDispatcher = {
```
Comment (elanalynn) [inline public/index.ts]: This could also be `const`

File: public/index.ts (hunk 1 above)
Comment (aprildagonese) [reply public/index.ts]: It was just leftover from some troubleshooting last week :) I changed it back.

File: public/index.ts (hunk 2 above)
Comment (aprildagonese) [reply public/index.ts]: Done!

File: public/index.ts (hunk 3 above)
Comment (aprildagonese) [reply public/index.ts]: Whoops, good catch!

File: public/index.ts (hunk 4 above)
Comment (aprildagonese) [reply public/index.ts]: Done.

**Pull request merged**